plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/Realtime.tsx

```diff
@@ -0,0 +1,150 @@
+import { Icon, Switch } from "@web/sharedComponents";
+import { styled } from "@web/theme";
+import { useCallback, useEffect, useRef, useState } from "react";
+
+import { BaseFieldProps } from "../types";
+
+const Realtime: React.FC<BaseFieldProps<"realtime">> = ({ value, editMode, onUpdate }) => {
+  const [intervalInSecond, setIntervalValue] = useState<number>(value.updateInterval ?? 30);
+  const [timer, setTimer] = useState<string>("00:00:00");
+  const [enableUpdate, changeUpdateState] = useState<boolean>(true);
+  const Ref = useRef<any>(null);
+
+  const getTimeRemaining = useCallback((interval: number) => {
+    const seconds = Math.floor(interval % 60);
+    const minutes = Math.floor((interval / 60) % 60);
+    const hours = Math.floor((interval / 60 / 60) % 24);
+    if (interval > 0) {
+      setTimer(
+        (hours > 9 ? hours : "0" + hours) +
+          ":" +
+          (minutes > 9 ? minutes : "0" + minutes) +
+          ":" +
+          (seconds > 9 ? seconds : "0" + seconds),
+      );
+    }
+  }, []);
+
+  const startTimer = useCallback(
+    (interval: number) => {
+      if (Ref.current) clearInterval(Ref.current);
+      const id = setInterval(() => {
+        if (enableUpdate) {
+          getTimeRemaining(interval);
+          interval--;
+          if (interval === 0) interval = intervalInSecond;
+        }
+      }, 1000);
+
+      Ref.current = id;
+    },
+    [enableUpdate, getTimeRemaining, intervalInSecond],
+  );
+  const clearTimer = useCallback(() => {
+    if (Ref.current) clearInterval(Ref.current);
+  }, []);
+
+  const handleChangeupdateState = useCallback(() => {
+    changeUpdateState(!enableUpdate);
+    setTimer("00:00:30");
+    if (enableUpdate) startTimer(intervalInSecond);
+  }, [enableUpdate, intervalInSecond, startTimer]);
+
+  const handleChangeUpdateTime = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setIntervalValue(parseInt(e.currentTarget.value));
+      onUpdate({
+        ...value,
+        updateInterval: parseInt(e.currentTarget.value),
+      });
+      startTimer(intervalInSecond);
+    },
+    [intervalInSecond, onUpdate, startTimer, value],
+  );
+
+  useEffect(() => {
+    if (enableUpdate) {
+      setTimer("00:00:30");
+      startTimer(intervalInSecond);
+    } else clearTimer();
+  }, [clearTimer, enableUpdate, intervalInSecond, startTimer]);
+
+  return editMode ? (
+    <Wrapper>
+      <Field>
+        <FieldTitle>Update time</FieldTitle>
+        <FieldValue>
+          <TextInput
+            readOnly={!enableUpdate}
+            value={intervalInSecond}
+            onChange={handleChangeUpdateTime}
+          />
+        </FieldValue>
+      </Field>
+    </Wrapper>
+  ) : (
+    <Wrapper>
+      <Field gap={10}>
+        <Icon icon="clock" size={24} />
```

Comment:
This is a static icon. What we want here is a dynamic icon that updates with the time. Does Antd have this in their icon set or loading section?

Reply:
Antd icons  have spinning property but for the custom icons don't I can try to add rotation property to icon , or we can change the icon and use existed icons in antd , what do you think ?

Reply:
@lavalse  hello , we need your support here , in realtime update I used static icon that you provided us with , but shouldn't be with animation ? if yes , would you please provide us the library you're using

Reply:
@nina992 Please rely on Slack for communication between core members. @lavalse MIGHT notice this message on github, but Slack is a much more reliable place for quick replies.

Reply:
sure I did mention red-san on slack